Riot/Modules/TabBar/TabBarCoordinator.swift

```diff
@@ -301,7 +301,7 @@ final class TabBarCoordinator: NSObject, TabBarCoordinatorType {
         
         // TODO: Remove Matrix session handling from the view controller
         // MXSession is opened before set to MXKAccount, wait for account change to be sure is set at a moment
-        if let matrixSession = userSession.matrixSession {
+        if let matrixSession = userSession.matrixSession, self.masterTabBarController.mxSessions.contains(matrixSession) == false {
```

Comment:
Is it really needed to add it here?
It is already added in `userSessionsServiceDidAddUserSession`.

Reply:
In fact we have no sanity checks when adding an MXSession to `[MasterTabBarController addMatrixSession]`. We can also move the check `MasterTabBarController`.